examples/ui-demo/src/state/index.tsx

```diff
@@ -0,0 +1,170 @@
+"use client";
+
+import { alchemyConfig, Config, queryClient } from "@/app/config";
+import { AuthCardHeader } from "@/components/shared/AuthCardHeader";
+import { AlchemyClientState } from "@account-kit/core";
+import {
+  AlchemyAccountProvider,
+  AlchemyAccountsUIConfig,
+  AuthType,
+  useUiConfig,
+} from "@account-kit/react";
+import {
+  createContext,
+  Dispatch,
+  PropsWithChildren,
+  SetStateAction,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { useStore } from "zustand";
+import {
+  createDemoStore,
+  generateClassesForRoot,
+  generateStylesForRoot,
+} from "./store";
+
+export type ConfigContextType = {
+  config: Config;
+  setConfig: (partial: Partial<Config>) => void;
+  nftTransfered: boolean;
+  setNFTTransfered: Dispatch<SetStateAction<boolean>>;
+};
+
+export const ConfigContext = createContext<ConfigContextType | undefined>(
+  undefined
+);
+
+export function useConfig(): ConfigContextType {
+  const configContext = useContext(ConfigContext);
+
+  if (!configContext) {
+    throw new Error("config context must be present in root");
+  }
+
+  return configContext;
+}
+
+function convertDemoConfigToUiConfig(config: Config): AlchemyAccountsUIConfig {
+  const sections: AuthType[][] = [[{ type: "email" }]];
+
+  if (config.auth.showPasskey) {
+    sections.push([{ type: "passkey" }]);
+  }
+
+  if (config.auth.showSocial && config.auth.addGoogleAuth) {
+    if (config.auth.showPasskey) {
+      sections
+        .at(-1)!
+        .push({ type: "social", authProviderId: "google", mode: "popup" });
+    } else {
+      sections.push([
+        { type: "social", authProviderId: "google", mode: "popup" },
+      ]);
+    }
+  }
+
+  if (config.auth.showExternalWallets) {
+    sections.push([
+      {
+        type: "external_wallets",
+        walletConnect: { projectId: "30e7ffaff99063e68cc9870c105d905b" },
+      },
+    ]);
+  }
+
+  return {
+    illustrationStyle: config.ui.illustrationStyle,
+    auth: {
+      sections,
+      addPasskeyOnSignup: config.auth.addPasskey && config.auth.showPasskey,
+      header: (
+        <AuthCardHeader
+          theme={config.ui.theme}
+          logoDark={config.ui.logoDark}
+          logoLight={config.ui.logoLight}
+        />
+      ),
+    },
+  };
+}
+
+export function ConfigContextProvider(
+  props: PropsWithChildren<{
+    initialConfig?: Config;
+    initialState?: AlchemyClientState;
+  }>
+) {
+  const storeRef = useRef<ReturnType<typeof createDemoStore>>();
+  if (!storeRef.current) {
+    storeRef.current = createDemoStore(props.initialConfig);
+  }
+
+  const { config, updateConfig: setConfig } = useStore(storeRef.current);
+  const [nftTransfered, setNFTTransfered] = useState(false);
```

Comment:
we might wanna move these into the demo store too, but then the `nftTransfered` will be persisted